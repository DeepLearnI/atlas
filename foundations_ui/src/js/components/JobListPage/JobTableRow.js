import React, { Component } from 'react';
import PropTypes from 'prop-types';
import StartCell from './cells/StartTimeCell';
import StatusCell from './cells/StatusCell';
import JobIDCell from './cells/JobIDCell';
import DurationCell from './cells/DurationCell';
import UserCell from './cells/UserCell';
import JobActions from '../../actions/JobListActions';
import CommonActions from '../../actions/CommonActions';

class JobTableRow extends Component {
  constructor(props) {
    super(props);
    this.state = {
      job: this.props.job,
      rowNumber: this.props.rowNumber,
    };
  }

  render() {
    const { job, rowNumber } = this.state;

    const isError = CommonActions.isError(job.status);

    return (
      <div className="job-table-row">
        <StartCell startTime={job.start_time} isError={isError} rowNumber={rowNumber} />
        <StatusCell status={job.status} isError={isError} rowNumber={rowNumber} />
        <JobIDCell jobID={job.job_id} isError={isError} rowNumber={rowNumber} />
        <DurationCell
          duration={JobActions.getDateDiff(job.start_time, job.completed_time)}
          isError={isError}
          rowNumber={rowNumber}
        />
        <UserCell user={job.user} isError={isError} rowNumber={rowNumber} />
      </div>
    );
  }
}

JobTableRow.propTypes = {
  job: PropTypes.object,
  rowNumber: PropTypes.number,
};

JobTableRow.defaultProps = {
  job: {},
  rowNumber: 0,
};

export default JobTableRow;
