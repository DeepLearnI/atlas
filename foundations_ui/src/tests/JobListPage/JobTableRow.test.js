import React from 'react';
import ReactDOM from 'react-dom';
import JobTableRow from '../../js/components/JobListPage/JobTableRow';
import { shallow, mount } from 'enzyme';
import configureTests from '../setupTests';

configureTests();

it('Shallow Renders Job Table Row', () => {
  const wrapper = shallow(<JobTableRow/>);
  expect(wrapper).toMatchSnapshot();
});
