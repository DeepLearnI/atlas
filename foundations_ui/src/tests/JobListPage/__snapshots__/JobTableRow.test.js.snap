// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Renders Job Table Row 1`] = `
<div
  className="job-table-row"
  onClick={[Function]}
  onKeydown={[Function]}
  role="presentation"
>
  <CancelJobCell
    job={
      Object {
        "status": "completed",
      }
    }
    onSuccessfullDeletion={[Function]}
  />
  <JobIDCell
    isError={false}
    jobID=""
    rowNumber={0}
  />
  <StartTimeCell
    expand={false}
    isError={false}
    rowNumber={0}
    startTime=""
    status="completed"
  />
  <StatusCell
    isError={false}
    rowNumber={0}
    status="completed"
  />
  <DurationCell
    days=""
    duration={0}
    hours=""
    isError={false}
    minutes=""
    rowNumber={0}
    seconds=""
  />
  <UserCell
    isError={false}
    rowNumber={0}
    user=""
  />
</div>
`;
